### PR TITLE
Scrape electoral history of ex members

### PR DESCRIPTION
**Pull request by ondenman**
This is a step towards scraping electoral data of members. (everypolitician/everypolitician-data#24566).

The `:electoral_history` of ExMember objects is populated by electoral history dates structured as a hash. For example:

```
{"Tenth Parliament"=>
  {"Co-opted"=>"2004-05-24", "Oath of Allegiance"=>"2004-05-24", "Dissolution of Parliament"=>"2008-02-04"},
 "Eleventh Parliament"=>
  {"Elected on"=>"2008-03-12",
   "Oath of Allegiance as Parliamentary Secretary for Health"=>"2008-03-12",
   "Oath of Allegiance as Member"=>"2008-05-10",
   "Oath of Allegiance as Minister for Health, the Elderly and Community Care"=>"2010-02-10",
   "Ceased as Minister"=>"2013-03-09"},
 "Twelfth Parliament"=>
  {"Elected on"=>"2013-03-13", "Oath of Allegiance"=>"2013-04-06", "Resignation from Parliament"=>"2015-11-03"}}
```

### CODE REVIEW

File: test/ex_member_page_test.rb [hunk 1]
```diff
@@ -1,20 +1,27 @@
 # frozen_string_literal: true
 require_relative './test_helper'
 require_relative '../lib/ex_member_page.rb'
+require_relative '../lib/decorators/formatted_dates'
+require_relative '../lib/electoral_history'
```
Comment (tmtmtmtm) [inline test/ex_member_page_test.rb]: you should really only be requiring things that the test itself is directly using. If you actually _do_ need to require these here, then something seems off.

File: test/ex_member_page_test.rb [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
 
 describe MemberPage do
-  around { |test| VCR.use_cassette('GeorgeVella', &test) }
```
Comment (tmtmtmtm) [inline test/ex_member_page_test.rb]: Why are you removing this test?

File: test/ex_member_page_test.rb [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
   end
 
   it 'should respond to electoral_history' do
     subject.must_respond_to :electoral_history
   end
+
+  it 'should have an end date for the twelfth term' do
+    subject.electoral_history['Twelfth Parliament']['Resignation from Parliament']
+           .must_equal '2015-11-03'
+  end
```
Comment (tmtmtmtm) [inline test/ex_member_page_test.rb:24]: this isn't really testing very much of what the PR claims to be adding. Let's talk in person about a better way of testing things like this.

File: lib/electoral_history.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class ElectoralHistory < Scraped::HTML
+  def to_h
```
Comment (tmtmtmtm) [inline lib/electoral_history.rb]: Overriding the `to_h` method of a `Scraped::HTML` subclass seems wrong. Why do you need to do this?

File: lib/decorators/formatted_dates.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class FormattedDates < Scraped::Response::Decorator
```
Comment (tmtmtmtm) [inline lib/decorators/formatted_dates.rb]: I think you need to add some documentation here to explain what this Decorator is doing.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]

**Comment by ondenman** [comment]
Addresses: https://github.com/everypolitician/everypolitician-data/issues/25279

**Comment by ondenman** [comment]
Closing in favour of: https://github.com/everypolitician-scrapers/malta-parliament/pull/12

**Pull request closed without merging**